dot/state/block.go

```diff
@@ -706,6 +711,7 @@ func (bs *BlockState) HandleRuntimeChanges(newState *rtstorage.TrieState, rt run
 		return fmt.Errorf("failed to update code substituted block hash: %w", err)
 	}
 
+	bs.notifyRuntimeUpdated(newVersion)
```

Comment:
`newVersion` is nil if the `!codeSubBlockHash.Equal(common.Hash{})` if block isn't entered, you can just move getting the version to right before here, since it's cached if it's already been fetched before

Reply:
```suggestion
        newVersion, err := rt.Version()
        //err check
	bs.notifyRuntimeUpdated(newVersion)
```

Reply:
good point, updated.